overlay/usr/lib/inithooks/bin/wordpress.py

```diff
@@ -4,65 +4,128 @@
+    if domain is None:
+        return ''
+    domain = domain.strip()
+    if domain == 'NONE':
+        return domain
+    if domain == ("http://"):
+        domain = domain[7:]
+    elif domain == ("https://"):
+        domain = domain[8:]
+    domain.strip("/")
+    if not domain:
+        return ''
+    if domain.startswith("http://") or domain.startswith("https://"):
+        return domain
+    else:
+        return f"https://{domain}"
+
 
 def main():
+    opts: list[tuple[str, str]] = []
+
     try:
         opts, args = getopt.gnu_getopt(sys.argv[1:], "h",
-                                       ['help', 'pass=', 'email='])
+                                       ['help', 'pass=', 'email=', 'domain='])
     except getopt.GetoptError as e:
         usage(e)
 
     password = ""
     email = ""
+    domain = ""
     for opt, val in opts:
         if opt in ('-h', '--help'):
             usage()
         elif opt == '--pass':
             password = val
         elif opt == '--email':
             email = val
+        elif opt == '--domain':
+            domain = val
 
-    if not password:
+    d = Dialog('TurnKey Linux - Configuration')
+    if 'd' not in locals():
         d = Dialog('TurnKey Linux - First boot configuration')
+
+    if not password:
         password = d.get_password(
             "Wordpress Password",
             "Enter new password for the Wordpress 'admin' account.")
 
     if not email:
-        if 'd' not in locals():
-            d = Dialog('TurnKey Linux - First boot configuration')
-
         email = d.get_email(
             "Wordpress Email",
             "Please enter email address for the Wordpress 'admin' account.",
             "admin@example.com")
 
     inithooks_cache.write('APP_EMAIL', email)
-    
+
+    if domain != 'NONE':
+        domain = process_domain(domain)
+        while True:
+            domain = d.get_input(
+                'WordPress domain (optional)',
+                "Enter domain to serve, if no protocol prefix, will assume"
+                " https. Enter 'NONE' to not set a domain",
+                DEFAULT_DOMAIN)
+            domain = process_domain(domain)
+            if not domain:
+                d.error("Domain must be set - or 'NONE'")
+            else:
+                break
+
+    inithooks_cache.write('APP_DOMAIN', domain)
+
+    assert password is not None
     hashpass = hashlib.md5(password.encode('utf8')).hexdigest()
 
     m = MySQL()
-    m.execute('UPDATE wordpress.wp_users SET user_email=\"%s\" WHERE user_nicename=\"admin\";' % email)
-    m.execute('UPDATE wordpress.wp_users SET user_pass=\"%s\" WHERE user_nicename=\"admin\";' % hashpass)
+    m.execute('UPDATE wordpress.wp_users'
+              ' SET user_email=\"%s\"'
+              ' WHERE user_nicename=\"admin\";' % email)
+    m.execute('UPDATE wordpress.wp_users'
+              ' SET user_pass=\"%s\"'
+              ' WHERE user_nicename=\"admin\";' % hashpass)
+
```

Comment:
Shouldn't be manually formatting values here, add the values as a second argument tuple instead (see bookstack inithook for an example of how).

Reply:
Thanks for the heads up on this. It's been fixed.